lib/ord.js

```diff
@@ -156,7 +150,19 @@ const fGetAPIResources = (csn, global,packageIds) => {
  * @param {Object} csn object
  * @returns {Array<object>} The Event Resources array.
  */
-const fGetEventResources = (csn, global,packageIds) => global.aEvents.map((srv) => fCreateEventResourceTemplate(srv, csn.definitions[srv], global, packageIds)).filter((resource) => resource !== null && resource !== undefined);
+const fGetEventResources = (csn, global, packageIds) => {
+    if(global.aEvents.length === 0) return []
+
+    const services = []
+    for (const srv of global.aServices) {
+        const hasEvents = global.aEvents.find((eventName) => eventName.startsWith(srv));
+        if (hasEvents) {
+            services.push(srv)
+        }
+    }
+
+    return services.map((srv) => fCreateEventResourceTemplate(srv, csn.definitions[srv], global, packageIds))
```

Comment:
```suggestion
    for (const serviceName of global.aServices) {
        const hasEvents = global.aEvents.find((eventName) => eventName.startsWith(serviceName));
        if (hasEvents) {
            services.push(serviceName)
        }
    }

    return services.map((serviceName) => fCreateEventResourceTemplate(serviceName, csn.definitions[serviceName], global, packageIds))
```
I think `srv` implies to serviceName. I would suggest use serviceNmae. I like `hasEvents`, is clear and good name. I found another problem, we don't have logger. In principle, for example line 154, if the event is empty, we return empty list, in line 164 we might also turn emtpy list if aServices is emtpy. To distinguish two conditions but same return value, we need logger to help us. Just an idea for the future.